assets/js/modules/ads/components/PAXEmbeddedApp.js

```diff
@@ -0,0 +1,230 @@
+				args,
+				return: callback,
+			} );
+
+			return callback;
+		};
+	}, [] );
+
+	const paxAppRef = useRef();
+
+	const elementID = useMemo( () => {
+		return `googlesitekit-pax-embedded-app-${ instanceID }`;
+	}, [ instanceID ] );
+
+	const paxConfig = useMemo( () => {
+		return {
+			...( global?._googlesitekitPAXConfig || {} ),
+			clientConfig: {
+				contentContainer: `#${ elementID }`,
+			},
+		};
+	}, [ elementID ] );
+
+	const paxServices = useMemo( () => {
+		return {
+			authenticationService: {
+				get: callbackWithArgs( 'authenticationService:get', {
+					accessToken:
+						global?._googlesitekitPAXConfig?.authAccess
+							?.oauthTokenAccess,
+					authuser: '0',
+				} ),
+				fix: callbackWithArgs( 'authenticationService:fix', {
+					retryReady: true,
+				} ),
+			},
+			businessService: {
+				getBusinessInfo: callbackWithArgs(
+					'businessService:getBusinessInfo',
+					{
+						businessName: siteName,
+						// Part of the API for PAX.
+						// eslint-disable-next-line sitekit/acronym-case
+						businessUrl: siteURL,
+					}
+				),
+				fixBusinessInfo: callbackWithArgs(
+					'businessService:fixBusinessInfo',
+					{
+						retryReady: true,
+					}
+				),
+			},
+			conversionTrackingService: {
+				getSupportedConversionLabels: callbackWithArgs(
+					'conversionTrackingService:getSupportedConversionLabels',
+					{
+						conversionLabels: [
+							'purchase',
+							'subscribe',
+							'add_to_cart',
+							'begin_checkout',
+							'book_appointment',
+							'contact',
+							'request_quote',
+							'phone_call_leads',
+							'get_directions',
+							'submit_lead_form',
+							'sign_up',
+						],
+					}
+				),
+			},
+			termsAndConditionsService: {
+				notify: callbackWithArgs(
+					'termsAndConditionsService:notify',
+					{}
+				),
+			},
+		};
+	}, [ callbackWithArgs, siteName, siteURL ] );
+
+	const launchPAXApp = useCallback( async () => {
+		try {
+			paxAppRef.current =
+				await global.google.ads.integration.integrator.launchGoogleAds(
+					paxConfig,
+					paxServices
+				);
+
+			onLaunch?.();
+		} catch ( error ) {
+			setLaunchError( error );
+			global.console.error(
+				'Google Ads Partner Experience Error:',
+				error
+			);
+		}
+
+		setIsLoading( false );
+	}, [ paxConfig, paxServices, onLaunch ] );
+
+	useInterval( () => {
+		if ( ! launchGoogleAdsAvailable || hasLaunchedPAXApp ) {
+			return;
+		}
+
+		if (
+			typeof global?.google?.ads?.integration?.integrator
+				?.launchGoogleAds === 'function'
+		) {
+			setLaunchGoogleAdsAvailable( true );
+		}
+	}, 50 );
+
+	useEffect( () => {
+		if ( launchGoogleAdsAvailable && ! hasLaunchedPAXApp ) {
+			setHasLaunchedPAXApp( true );
+
+			launchPAXApp();
+		}
+	}, [
+		hasLaunchedPAXApp,
+		isLoading,
+		launchGoogleAdsAvailable,
+		launchPAXApp,
+	] );
+
+	return (
+		<div className="googlesitekit-pax-embedded-app">
+			{ !! launchError && ! isAdBlockerActive && (
```

Comment:
Should we add additional message for the scenario when `sAdBlockerActive` is active? We can include the `Ads` specific `AdBlockerWarning` [Component](https://github.com/google/site-kit-wp/blob/develop/assets/js/modules/ads/components/common/AdBlockerWarning.js)

Reply:
The PAX app itself has a large Ad Blocker warning message when an Ad Blocker is detected, so I thought it was best not to duplicate it as it's quite strange-looking and duplicates things.

We could in theory not render the PAX app at all when we detect an Ad Blocker, but I thought it was better to render their app than not, when we can 🤔

Reply:
I see, nice, seems better to let it handle the Ad blocker "natively" instead of us interfering it. Thanks